crates/matrix-sdk-ui/src/room_list_service/room_list.rs

```diff
@@ -129,6 +131,34 @@ impl RoomList {
             entries_stream.batch_with(self.room_list_service_state.clone().map(|_| ())),
         )
     }
+
+    /// Get a stream of room list entries, filtered dynamically.
+    ///
+    /// The returned stream will only start yielding diffs once a filter is set
+    /// through the returned `DynamicRoomListFilter`. For every call to
+    /// [`DynamicRoomListFilter::set`], the stream will yield a
+    /// [`VectorDiff::Reset`] followed by any updates of the room list under
+    /// that filter (until the next reset).
+    pub fn entries_with_dynamic_filter(
+        &self,
+    ) -> (impl Stream<Item = Vec<VectorDiff<RoomListEntry>>>, DynamicRoomListFilter) {
+        // FIXME: Should ideally be a single-consumer watch channel / observable.
+        let (tx, rx) = mpsc::channel(1);
```

Comment:
Why not using an `eyeball::Observable` directly?

Reply:
Because `mpsc::Receiver` gives ownership of the item being sent over the channel without cloning, which is required unless we require the filter closure to be `Clone`.